### PR TITLE
Feature/model list filter

### PR DESCRIPTION
**Pull request by JozefFlakus**
## Feature
- New `model-list` component visual implementation

## Improvements
- `model-table` based on material CDK (component development kit)
- dedicated `model-filter` component
- removed default `box` spacing
- redefined material theming settings

zrzut ekranu 2018-01-18 o 22 09 19

### CODE REVIEW

File: src/app/+models/components/model-table/model-table.component.html [hunk 1]
```diff
@@ -1,71 +1,71 @@
...
+    </cdk-header-cell>
+    <cdk-cell class="table__cell" *cdkCellDef="let element">
       <button
-        class="model-table__button"
         mat-icon-button
-        matTooltip="Edit model #{{ element.number }}"
-        [matTooltipPosition]="'before'"
-        (click)="$event.stopPropagation(); this.onEdit(element)"
+        [matMenuTriggerFor]="actionMenu"
+        (click)="$event.stopPropagation();"
```
Comment (sebastianmusial) [inline src/app/+models/components/model-table/model-table.component.html]: `;` in single action is unnecessary

File: src/app/+models/components/model-filter/model-filter.component.scss [hunk 2]
```diff
@@ -1 +1,20 @@
...
+
+.model-filter__icon {
+  @extend %u-text--icon-medium;
+  vertical-align: middle;
+  user-select: none;
+}
+
+.model-filter__field {
+  width: 100%;
+  padding-left: 54px;
```
Comment (sebastianmusial) [inline src/app/+models/components/model-filter/model-filter.component.scss]: `54px` - this value should be always the same or maybe we should use our global dimensions?

File: src/app/+models/components/model-table/model-table.component.scss [hunk 3]
```diff
@@ -1,19 +1,74 @@
...
-  filter: grayscale(100%);
+.table__header-cell {
+  flex: 1 0 0;
+  font-weight: bold;
+
+  &:first-child {
+    flex: 0 0 auto;
+    width: 100%;
+    max-width: 25px;
+    margin-right: 50px;
```
Comment (sebastianmusial) [inline src/app/+models/components/model-table/model-table.component.scss]: `50px` - this value should be always the same or maybe we should use our global dimensions?

File: src/app/+models/components/model-table/model-table.component.scss [hunk 4]
```diff
@@ -1,19 +1,74 @@
...
+.table__cell {
+  display: flex;
+  flex: 1 0 0;
+  align-items: center;
+
+  &:first-child {
+    flex: 0 0 auto;
+    width: 100%;
+    max-width: 25px;
+    margin-right: 50px;
```
Comment (sebastianmusial) [inline src/app/+models/components/model-table/model-table.component.scss]: `50px` - this value should be always the same or maybe we should use our global dimensions?

File: src/app/+models/components/model-filter/model-filter.component.scss (hunk 2 above)
Comment (JozefFlakus) [reply src/app/+models/components/model-filter/model-filter.component.scss]: it is dedicated only for this class

File: src/app/+models/components/model-table/model-table.component.scss (hunk 4 above)
Comment (JozefFlakus) [reply src/app/+models/components/model-table/model-table.component.scss]: always the same - it shouldn't interpolate - it is related to spacing inside `model-filter.component`

File: src/app/+models/components/model-table/model-table.component.scss (hunk 3 above)
Comment (JozefFlakus) [reply src/app/+models/components/model-table/model-table.component.scss]: always the same - it shouldn't interpolate - it is related to spacing inside `model-filter.component`

**Pull request merged**